router/transformer/transformerProxyAdapter.go

```diff
@@ -0,0 +1,152 @@
+package transformer
+
+//go:generate mockgen -destination=../../mocks/router/transformer/mock_transformer.go -package=mocks_transformer github.com/rudderlabs/rudder-server/router/transformer Transformer
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+	"github.com/rudderlabs/rudder-go-kit/stats"
+	"github.com/rudderlabs/rudder-server/processor/integrations"
+)
+
+type transformerProxyAdapter interface {
+	getPayload(proxyReqParams *ProxyRequestParams) ([]byte, error)
+	getProxyURL(destType string) (string, error)
+	getResponse(response []byte, respCode int, metadata []ProxyRequestMetadata) (TransResponseT, error)
+}
+
+type ProxyRequestPayloadV0 struct {
+	integrations.PostParametersT
+	Metadata ProxyRequestMetadata `json:"metadata"`
+}
+
+type ProxyResponseV0 struct {
```

Comment:
if more versions came then we need to write adapters for every combination?

Reply:
We need an adapter per version.